neo/public/javascripts/map/controllers.js

```diff
@@ -394,56 +397,94 @@ angular.module('cloudberry.map', ['leaflet-directive', 'cloudberry.common'])
             d.properties.count = 0;
+          if (d.properties.countText)
+            d.properties.countText = "";
           for (var k in result) {
           //TODO make a hash map from ID to make it faster
-            if (result[k].state == d.properties.stateID) {
-              d.properties.count = result[k].count;
+            if (result[k].state === d.properties.stateID) {
+              if($scope.doNormalization){
+                d.properties.count = result[k]['count'] / result[k]['population'] * cloudberryConfig.normalizationUpscaleFactor;
+                setNormalizedPropCountText(d);
+              }
+              else{
+                d.properties.count = result[k]['count'];
+                d.properties.countText = d.properties.count.toString();
+              }
             }
           }
         });
 
         // draw
         $scope.polygons.statePolygons.setStyle(style);
 
-      } else if ($scope.status.logicLevel == "county" && $scope.geojsonData.county) {
+      } else if ($scope.status.logicLevel === "county" && $scope.geojsonData.county) {
           angular.forEach($scope.geojsonData.county.features, function(d) {
             if (d.properties.count)
               d.properties.count = 0;
+            if (d.properties.countText)
+              d.properties.countText = "";
             for (var k in result) {
               //TODO make a hash map from ID to make it faster
-              if (result[k].county == d.properties.countyID) {
-                d.properties.count = result[k].count;
+              if (result[k].county === d.properties.countyID) {
+                if($scope.doNormalization){
+                  d.properties.count = result[k]['count'] / result[k]['population'] * cloudberryConfig.normalizationUpscaleFactor;
+                  setNormalizedPropCountText(d);
+                }
+                else{
+                  d.properties.count = result[k]['count'];
+                  d.properties.countText = d.properties.count.toString();
+                }
               }
             }
           });
 
         // draw
         $scope.polygons.countyPolygons.setStyle(style);
 
-      }else if ($scope.status.logicLevel == "city" && $scope.geojsonData.city) {
+      }else if ($scope.status.logicLevel === "city" && $scope.geojsonData.city) {
         angular.forEach($scope.geojsonData.city.features, function(d) {
           if (d.properties.count)
             d.properties.count = 0;
+          if (d.properties.countText)
+            d.properties.countText = "";
           for (var k in result) {
             //TODO make a hash map from ID to make it faster
-            if (result[k].city == d.properties.cityID) {
-              d.properties.count = result[k].count;
+            if (result[k].city === d.properties.cityID) {
+              if($scope.doNormalization){
+                d.properties.count = result[k]['count'] / result[k]['population'] * cloudberryConfig.normalizationUpscaleFactor;
```

Comment:
could you define a util function like `normalizeCount` to make the three places of the same code look better?